google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -299,6 +303,40 @@ int getMessageDeadlineSeconds() {
     return messageDeadlineSeconds.get();
   }
 
+  @InternalApi
+  void setEnableExactlyOnceDelivery(boolean enableExactlyOnceDelivery) {
+    // Sanity check that we are changing the enableExactlyOnceDelivery state and that we are using
+    // the default min
+    if (enableExactlyOnceDelivery == this.enableExactlyOnceDelivery.get()
+        || !minDurationPerAckExtensionDefaultUsed) {
+      return;
+    }
+
+    // We just need to update the minDurationPerAckExtensionSeconds as the
+    // maxDurationPerAckExtensionSeconds does not change
+    int possibleNewMinAckDeadlineExtensionSeconds;
+
+    if (enableExactlyOnceDelivery) {
```

Comment:
In this case, do you want to check if the user has provided a min ack deadline? If they have, we should use the user provided ack deadline. Same in the `else` case below, instead of using defaults, right?

Reply:
Above we are checking if the user has provided a min ack deadline and then returning.
```
if (enableExactlyOnceDelivery == this.enableExactlyOnceDelivery.get()
        || !minDurationPerAckExtensionDefaultUsed) {
      return;
    }
```